src/software/ai/hl/stp/play/defense/defense_play_base.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "proto/parameters.pb.h"
+#include "shared/constants.h"
+#include "software/ai/hl/stp/play/play_fsm.h"
+#include "software/ai/hl/stp/tactic/crease_defender/crease_defender_tactic.h"
+#include "software/ai/hl/stp/tactic/pass_defender/pass_defender_tactic.h"
+#include "software/ai/evaluation/defender_assignment.h"
+
+#include "software/logger/logger.h"
+
+/**
+ * Struct containing frequently shared functions of the defense play class
+ */
+class DefensePlayBase {
+
+public:
+    struct ControlParams
+    {
+        // The maximum allowed speed mode
+        TbotsProto::MaxAllowedSpeedMode max_allowed_speed_mode;
+    };
+
+    DEFINE_PLAY_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    /**
+     * Creates a play FSM with defensive methods
+     *
+     * @param ai_config the play config for this play FSM
+     */
+    explicit DefensePlayBase(TbotsProto::AiConfig ai_config);
+
+    /**
+     * Helper function to set up crease defender tactic vector members
+     *
+     * @param num_crease_defenders the number of crease defender tactics to set
+     */
+    void setUpCreaseDefenders(unsigned int num_crease_defenders);
+
+    /**
+     * Helper function to set up pass defender tactic vector members
+     *
+     * @param num_pass_defenders the number of pass defender tactics to set
+     */
+    void setUpPassDefenders(unsigned int num_pass_defenders);
```

Comment:
I'd prefer not to use `unsigned`. let's use a regular `int`. I don't want someone calling this function accidentally and underflow the argument into this function